gum/backend-windows/gumprocess-windows.c

```diff
@@ -309,9 +311,71 @@ void
 gum_process_enumerate_malloc_ranges (GumFoundMallocRangeFunc func,
                                      gpointer user_data)
 {
-  /* Not implemented */
-  (void) func;
-  (void) user_data;
+  HANDLE process_heap;
+  DWORD num_heaps;
+  HANDLE * heaps;
+  DWORD i;
+
+  process_heap = GetProcessHeap ();
+  if (!gum_process_enumerate_heap_ranges (process_heap, func, user_data))
+    return;
+
+  num_heaps = GetProcessHeaps (0, NULL);
+  if (num_heaps == 0)
+    return;
+  heaps = HeapAlloc (process_heap, 0, num_heaps * sizeof (HANDLE));
+  if (heaps == NULL)
+    return;
+  num_heaps = GetProcessHeaps (num_heaps, heaps);
+  if (num_heaps == 0)
+    goto beach;
+
+  for (i = 0; i != num_heaps; i++)
+  {
+    if (heaps[i] != process_heap)
+    {
+      if (!gum_process_enumerate_heap_ranges (process_heap, func, user_data))
+        break;
+    }
+  }
+
+beach:
+  HeapFree (process_heap, 0, heaps);
+
+  return;
```

Comment:
This `return` is redundant. (So we can then also remove the blank line above it.)